js/coinflex.js

```diff
@@ -0,0 +1,1885 @@
+        //         "remainQuantity": "0.0", // field not present if order doesn't have any fills
+        //         "notice": "OrderMatched",
+        //         "orderType": "LIMIT",
+        //         "fees": "0.003857920", // field not present if order doesn't have any fills
+        //         "feeInstrumentId": "USD", // field not present if order doesn't have any fills
+        //         "isTriggered": "false"
+        //     }
+        //
+        const marketId = this.safeString (order, 'marketCode');
+        market = this.safeMarket (marketId, market);
+        const isCreateOrder = 'timestamp' in order;
+        const symbol = market['symbol'];
+        const timestamp = this.safeInteger2 (order, 'timestamp', 'orderCreated');
+        const orderId = this.safeString (order, 'orderId');
+        const clientOrderId = this.safeString (order, 'clientOrderId');
+        const statusRaw = this.safeString (order, 'status');
+        const status = this.parseOrderStatus (statusRaw);
+        const sideRaw = this.safeString (order, 'side');
+        const side = this.parseOrderSide (sideRaw);
+        const orderTypeRaw = this.safeString (order, 'orderType');
+        const orderType = this.parseOrderType (orderTypeRaw);
+        let filledQuantityRaw = this.safeString (order, 'filledQuantity');
+        if (isCreateOrder) {
+            filledQuantityRaw = this.safeString (order, 'matchQuantity', '0');
+        }
+        const avgPriceRaw = this.safeString (order, 'avgFillPrice');
+        const timeInForceRaw = this.safeString (order, 'timeInForce');
+        const timeInForce = this.parseTimeInForce (timeInForceRaw);
+        const price = this.safeString (order, 'price');
+        const limitPrice = this.safeString (order, 'limitPrice');
+        let finalLimitPrice = undefined;
+        if (isCreateOrder) {
+            finalLimitPrice = price;
+        } else {
+            finalLimitPrice = limitPrice;
+        }
+        let trades = this.safeValue (order, 'matchIds');
+        if (trades !== undefined) {
+            trades = this.parseTrades (trades, market);
+        }
+        let cost = undefined;
+        if (avgPriceRaw !== undefined && filledQuantityRaw !== undefined) {
+            cost = Precise.stringMul (avgPriceRaw, filledQuantityRaw);
+        }
+        const feesRaw = this.safeValue (order, 'fees');
+        let fees = undefined;
+        if (feesRaw !== undefined) {
+            if (isCreateOrder) {
+                fees = {
+                    'currency': this.safeString (order, 'feeInstrumentId'),
+                    'fee': this.safeString (order, 'fees'),
+                };
+            } else {
+                const feeKeys = Object.keys (feesRaw);
+                if (feeKeys.length > 0) {
+                    const firstKey = feeKeys[0];
+                    fees = {
+                        'currency': firstKey,
```

Comment:
aren't we missing a `this.safeCurrencyCode(firstKey)` here?

Reply:
good catch!